includes/class-wc-stripe-helper.php

```diff
@@ -443,6 +443,25 @@ public static function get_order_by_setup_intent_id( $intent_id ) {
 		return false;
 	}
 
+	/**
+	 * Sanizite and retrieve a shortened statement descriptor.
+	 *
+	 * The descriptor will be used as prefix and the order ID will be concatenated, acting as a suffix.
+	 *
+	 * @param string $statement_descriptor
+	 * @param WC_Order $order
+	 * @return string $statement_descriptor Final shortened statement descriptor
+	 */
+	public static function get_shortened_statement_descriptor( $statement_descriptor = '', $order = null ) {
+		$statement_descriptor = self::clean_statement_descriptor( $statement_descriptor );
+
+		if ( method_exists( $order, 'get_id' ) && ! empty( $order->get_id() ) ) {
+			$statement_descriptor = $statement_descriptor . '* #' . $order->get_id();
```

Comment:
Are we concerned at all about the concatenated statement descriptor being too long? If the static part is up to 13 characters, that leaves 9 for the order number (if that's what we use instead of ID), which seems like plenty for sequential IDs but might not be enough for all cases. Should we truncate it to avoid an error?

Reply:
Good point, done in 9446e3d.